github/project.go

```diff
@@ -73,14 +73,19 @@ func (p *Project) GitURL(name, owner string, isSSH bool) (url string) {
 	host := rawHost(p.Host)
 
 	if preferredProtocol() == "https" {
-		url = fmt.Sprintf("https://%s/%s/%s.git", host, owner, name)
-	} else if isSSH || preferredProtocol() == "ssh" {
-		url = fmt.Sprintf("git@%s:%s/%s.git", host, owner, name)
-	} else {
-		url = fmt.Sprintf("git://%s/%s/%s.git", host, owner, name)
+		return fmt.Sprintf("https://%s/%s/%s.git", host, owner, name)
 	}
 
-	return url
+	if isSSH || preferredProtocol() == "ssh" {
+		return fmt.Sprintf("git@%s:%s/%s.git", host, owner, name)
+	}
+
+	if preferredProtocol() == "git" {
```

Comment:
If someone specifies `git` as their preferred protocol, does that still use `ssh` for "writeable" repos, e.g. cloning a repo to which you have writeable permissions? (This is due `git` protocol not being pushable.)

This would be a way people could opt into the old behavior before this change, but this behavior doesn't yet seem exercised in depth by updates to tests.

Reply:
Nope, if `git` is specified as the preferred protocol, all repos cloned will use that protocol regardless of their repo privileges/privacy. Wouldn't it be appropriate to respect that setting and clone using the plain `git` URL?

> This would be a way people could opt into the old behavior before this change, but this behavior doesn't yet seem exercised in depth by updates to tests.

I'm not clear on what you mean. Are you referring to people who currently use `hub`, and currently set `hub.protocol` to `git`, and not surprising them with this change in behavior?

Reply:
Hmm, okay, that sounds fair. This change should be made inside `clone.go`, right? Something like this?

```
				if !isSSH &&	
					args.Command != "submodule" &&	
					github.IsGitProtocol() {	
					isSSH = repo.Private || repo.Permissions.Push
```